### DISCUSSION TITLE
Development

### DISCUSSION THREAD

**Discussion by NikZak** [smartcorelib/smartcore#177] [Ideas]
There is no section about development. The library looks quite interesting. And recent. And the docs are nicely made. 
But it is not clear whether the team welcomes contributions from outside. E.g. among all tree-based ensemble classifiers the ExtraTreesClassifier usually serves me best. Here we have RandomForestClassifier only. I love Rust and would probably take the challenge of implementing a well parallelized version of ExtraTreesClassifier (and maybe regressor) but it is not clear whether the team would welcome such contribution. I suggest adding a few lines about development contribs in the readme file of the repo.

**Comment by Mec-iS** [comment]
Hi, you are perfectly right.

Contributions are welcome in accordance to license and common Github code of conduct.
We have not had time to care about the developer experience side of things unfortunately, I will try to add a code of conduct and some templates for opening PRs and issues.

**Comment by Mec-iS** [comment]
see #178

**Comment by Mec-iS** [comment]
We now have contribution guidelines

**Reply by NikZak** [reply]
Great work! Thank you!

**Comment by NikZak** [comment]
I would also suggest a place with a list of features so that a new developer might come and pick one (maybe Trello, maybe Projects tab in this repo or maybe just an issue with a list).
Among things that I usually use I can think of simple imputer (median, mean), robust scaler (I saw standard scaler only), imbalanced learning data preprocessors, feature selectors, pipeline (similar to sklearn but that might require architectural changes) and so on. Sorry if some of that is already implemented. Meanwhile, I guess new features can be ad hoc discussed at the discussion board

**Reply by Mec-iS** [reply]
usually technical discussions take place in comments so they can be kept with the code for future reference.

**Comment by morenol** [comment]
We have right now several issues with the label `help-wanted`. You can filter the issues with that label. What do you think? Is that enough?

**Reply by Mec-iS** [reply]
yes exactly. anybody can open issues and tag them accordingly to how newcomer-friendly they are.

**Comment by Mec-iS** [comment]
~I will also add a `good-first-issue` or `good-for-newcomers` label for issues~
there is a `good fist issue` label that we should add to the issues
